packages/cardpay-cli/scheduled-payment/create-sp-hash.ts

```diff
@@ -0,0 +1,126 @@
+import { Argv } from 'yargs';
+import { getSDK } from '@cardstack/cardpay-sdk';
+import { getEthereumClients, getConnectionType, NETWORK_OPTION_ANY } from '../utils';
+import { Arguments, CommandModule } from 'yargs';
+
+export default {
+  command:
+    'create-sp-hash <moduleAddress> <tokenAddress> <amount> <payeeAddress> <fixedUSDFee> <percentageFee> <executionGas> <maxGasPrice> <gasTokenAddress> <salt> <payAt>',
+  describe: 'Create scheduled payment hash',
+  builder(yargs: Argv) {
+    return yargs
+      .positional('moduleAddress', {
+        type: 'string',
+        description: 'The address of scheduled payment module',
+      })
+      .positional('tokenAddress', {
+        type: 'string',
+        description: 'The address of the token being transferred',
+      })
+      .positional('amount', {
+        type: 'string',
+        description: 'Amount of tokens you would like transferred (in the smallest units of token)',
+      })
+      .positional('payeeAddress', {
+        type: 'string',
+        description: 'The address of the payee of scheduled payment',
+      })
+      .positional('fixedUSDFee', {
+        type: 'number',
+        description: 'Fixed USD fee (e.g. 0.25)',
+      })
+      .positional('percentageFee', {
+        type: 'number',
+        description: 'Percentage fee (e.g. 5%, 0.05)',
+      })
+      .positional('percentageFee', {
+        type: 'number',
+        description: 'Percentage fee (e.g. 5%, 0.05)',
+      })
+      .positional('executionGas', {
+        type: 'number',
+        description: 'The gas limit to execute scheduled payment',
+      })
+      .positional('maxGasPrice', {
+        type: 'string',
+        description: 'Maximum gas price (in the smallest units of gas token)',
+      })
+      .positional('gasTokenAddress', {
+        type: 'string',
+        description: 'The address of the gas token',
+      })
+      .positional('salt', {
+        type: 'string',
+        description: 'Arbitrary string to make SP unique',
+      })
+      .positional('payAt', {
+        type: 'number',
+        description: 'Time to execute scheduled payments (in seconds)',
+      })
+      .option('recurringDayOfMonth', {
+        type: 'number',
+        description: 'Days of the month in the range of 1-28 to make recurring payments',
```

Comment:
Maybe better: "Day of the month on which the payment will be made recurringly (range: 1-28)"

Reply:
maybe we can use 1-31, and then we could on the server use moment.js (or your favorite date tool) to detect how many days there are in the current current month, and if the number provided by the user is larger than the current day, and the current day is the last day of the month, then we could execute the payment. And the way we could explain this is that we accept a range between 1 - 31, and that for months with less days that the specified value, we'll execute the payment on the last day of the month. (I generally have bills that I like to schedule to be paid on the last day of the month--it's probably a popular option actually).